tools/calculate-release/CalculateRelease.hs

```diff
@@ -1,4 +1,66 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-}
+
 module CalculateRelease(main) where
 
+import Control.Lens
+import Control.Monad
+import Data.Aeson.Lens
+import qualified Data.ByteString.Lazy.Char8 as LBSC
+import Data.Time
+
+import Network.Wreq
+import System.Exit
+import Text.Regex.TDFA
+
+import Chainweb.BlockHeight
+
+main :: IO ()
 main = do
-    read <$> interact "Latest block height: "
+    -- fetch current service date
+    nodeMainModule <- get "https://raw.githubusercontent.com/kadena-io/chainweb-node/master/node/ChainwebNode.hs" <&> (^. responseBody . to LBSC.unpack)
+    let serviceDateRegex :: String = "^\\s*serviceDate = Just \"([0-9]+-[0-9]+-[0-9]+)(T[0-9:]+Z)?\""
+    let serviceDateMatch :: (String, String, String, [String]) =
+            (nodeMainModule =~ serviceDateRegex)
+    serviceDateDay <- parseTimeM True defaultTimeLocale "%Y-%-m-%-d" $
+        (serviceDateMatch ^?! _4 . _head)
+    putStrLn $ "Current service date detected: " <> show serviceDateDay
+    let serviceDateTime = UTCTime { utctDay = serviceDateDay, utctDayTime = 0 }
+
+    -- check current time difference from service date
+    now <- getCurrentTime
+    when (serviceDateTime `diffUTCTime` now > 30 * 24 * 60 * 60) $
+        putStrLn $ unlines
+            [ "WARNING: estimating the fork height from >30 days in advance."
+            , "This may result in an inaccurate fork height. Remember that an"
+            , "early fork height can fork the network, and a late fork height can"
+            , "result in team members not being around during the fork." ]
+
+    !mainnetHeightNow <- heightOfChain0 "https://us-e1.chainweb.com/chainweb/0.0/mainnet01/cut"
+    mainnetTimeNow <- getCurrentTime
+    !testnetHeightNow <- heightOfChain0 "https://us1.testnet.chainweb.com/chainweb/0.0/testnet04/cut"
+    testnetTimeNow <- getCurrentTime
+    let mainnetForkTime = addUTCTime (24 * 60 * 60) serviceDateTime
+    let testnetForkTime = addUTCTime (12 * 60 * 60) serviceDateTime
```

Comment:
We may consider making this configurable -- but it's fine for now. We haven't changed this in a long time.